tests/interpreters/test_ml_program_interpreter.py

```diff
@@ -35,7 +35,7 @@ def test_ml_program_global_load_constant():
     interpreter.register_implementations(MLProgramFunctions())
 
     (result,) = interpreter.run_op(fetch, ())
-    assert result == ShapedArray(TypedPtr.new_int32([4]), [4])
+    assert result == ShapedArray(TypedPtr.new_int32([4] * 4), [4])
```

Comment:
Same here, why is the interface changing?

Reply:
In this case I do think the test was a bit fishy.
If you ask the values of a `1x4` array, would you not expect for the array to have 4 values (regardless of whether it is initialized as a splat)?
The previous implementation allowed for the nb of stored elements to differ from the nb of elements defined by the shape. This new one doesn't which is why

Reply:
resolved in #3637